pkg/controller/migration/convert/core.go

```diff
@@ -207,6 +205,36 @@ func handleCore(c *components, install *Installation) error {
 	return nil
 }
 
+// addResources adds the rescReq resource for the specified component if none was previously set. If installation
+// already had a resource for compName then they are compared and if they are different then an error is returned.
+// If the Resource is added to installation or the existing one matches then nil is returned.
+func addResources(install *Installation, compName operatorv1.ComponentName, rescReq *corev1.ResourceRequirements) error {
+	if install.Spec.ComponentResources == nil {
+		install.Spec.ComponentResources = []*operatorv1.ComponentResource{}
+	}
+
+	var existingRR *corev1.ResourceRequirements
+	// See if there is already a ComponentResource with the name
+	for _, x := range install.Spec.ComponentResources {
+		if x.ComponentName == compName {
+			existingRR = x.ResourceRequirements
+			break
+		}
+	}
+	if existingRR == nil {
+		install.Spec.ComponentResources = append(install.Spec.ComponentResources, &operatorv1.ComponentResource{
+			ComponentName:        compName,
+			ResourceRequirements: rescReq.DeepCopy(),
+		})
+		return nil
+	}
+	if reflect.DeepEqual(existingRR, rescReq) {
+		return nil
+	}
+
+	return ErrIncompatibleCluster{fmt.Sprintf("ResourcesRequirements for component %s did not match between Installation and migration source", compName)}
```

Comment:
Maybe in the future we should just prefer whatever is set in the `Installation` but this seems ok for now while the "user-specified install" use case is still not supported